LibraBFT/Impl/Consensus/BlockStorage/BlockStore.agda

```diff
@@ -34,6 +48,44 @@ executeAndInsertBlockM b = do
       put (rmSetBlockStore s bs')
       ok eb
 
+executeAndInsertBlockE bs0 block =
+  maybeS (getBlock (block ^∙ bId) bs0) continue (pure ∘ (bs0 ,_))
+ where
+  continue : Either ErrLog (BlockStore _ × ExecutedBlock)
+  continue =
+    maybeS (bs0 ^∙ bsRoot _) (Left fakeErr) λ bsr →
+    let btRound = bsr ^∙ ebRound in
+    if-dec btRound ≥?ℕ block ^∙ bRound
+    then Left fakeErr -- block with old round
+    else do
+      eb ← case executeBlockE bs0 block of λ where
+        (Right res) → Right res
+        (Left (ErrBlockNotFound parentBlockId)) → do
+          eitherS (pathFromRoot parentBlockId bs0) Left $ λ blocksToReexecute →
```

Comment:
Curious why this is different than the Haskell code.  We have `maybeS`, right?

Reply:
Two things:
- this looks just like the Haskell code to me
- working eith `Either` here, so `maybeS` not applicable.